### PR TITLE
task: project structure

### PR DESCRIPTION
**Pull request by OlegMakarenko**
**What's the change**
- Moved package.json and configs from `packages/paper-wallets` to the project root.
- Added linter and pre-commit hook.
- Updated README.
- Updated dependencies.
- Added Jenkinsfile.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "contributors": [
+        {
+            "name": "Peersyst Technology",
+            "email": "dev@peersyst.com"
+        },
+        {
+            "name": "NEM Software Ltd",
+            "email": "info@nem.software"
+        }
+    ],
```
Comment (AnthonyLaw) [inline package.json]: This is a new project, right?
If yes I don't think we need this.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/symbol/wallets-lib.git"
+    },
+    "homepage": "https://github.com/symbol/wallets-lib#readme",
+    "dependencies": {
+        "@pdf-lib/fontkit": "1.0.0",
+        "pdf-lib": "1.10.0",
+        "symbol-qr-library": "^0.13.1-alpha-202010241730"
```
Comment (AnthonyLaw) [inline package.json]: It needs to update to the latest version v0.14.2

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "semi": true,
+    "trailingComma": "all",
+    "singleQuote": true,
+    "tabWidth": 4
+}
```
Comment (AnthonyLaw) [inline .prettierrc]: I think most of our project was using code length 140.
let add `printWidth: 140`

**Review by AnthonyLaw — COMMENTED** [review]
All look good for me.
added some comments.

File: README.md [hunk 4]
```diff
@@ -1,26 +1,25 @@
-# Symbol Paper Wallets
-
-Symbol paper wallet generator
+# Symbol Wallets Lib
 
 1. [Installation](#installation)
-2. [Usage](#usage)
+2. [Paper Wallet Usage](#paper-wallet-usage)
```
Comment (AnthonyLaw) [inline README.md:6]: It may be good to add *requirement* such as node version

File: package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        "@typescript-eslint/eslint-plugin": "^4.10.0",
+        "@typescript-eslint/parser": "^4.9.0",
+        "rxjs": "^6.5.4",
+        "eslint": "^7.15.0",
+        "eslint-config-prettier": "^6.15.0",
+        "eslint-plugin-header": "^3.1.0",
+        "eslint-plugin-prettier": "^3.2.0",
+        "prettier": "^2.2.1",
+        "prettier-plugin-organize-imports": "^1.1.1",
+        "typescript": "^4.1.2"
```
Comment (yilmazbahadir) [inline package.json]: Are these dependencies up to date? Could you run the command below and update deps if necessary?
```
npm outdated
```

**Comment by yilmazbahadir** [comment]
when running `build` it doesn't transpile files into `dist` directory. In order to do that and copy the js files(from resources dir), you should make the following **_changes_** in `tsconfig.json`


{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es2016",
    "noImplicitAny": false,
    "strictNullChecks": true,
    "preserveConstEnums": true,
    "experimentalDecorators": true,
    "sourceMap": true,
    "declaration": true,
    "moduleResolution": "node",
    "stripInternal": true,
    _**"allowJs": true,**_
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "lib": [
      "es2016",
      "dom"
    ],
    "typeRoots": [
      "node_modules/@types"
    ],
    "noStrictGenericChecks": true,
    _**"outDir": "./dist/",**_
  },
  "files": [
    "index.ts"
  ],
  "exclude": [
    "node_modules",
    "build"
  ],
  "include": [
    "src/**/*.ts"
  ]
}

**Comment by yilmazbahadir** [comment]
And also do we have any `Jenkinsfile` where we have the lint, test, build and publish steps? Should we put some info into `Changelog` when releasing?

File: package.json (hunk 1 above)
Comment (OlegMakarenko) [reply package.json]: This is from `symbol-paper-wallets` project. Now it is part of `wallets-lib`

**Comment by OlegMakarenko** [comment]
@yilmazbahadir 

> "allowJs": true,

Thanks, missed that.

>"outDir": "./dist/",

Yes, will rename `build` into `dist`.

>And also do we have any Jenkinsfile where we have the lint, test, build and publish steps?

No we dont. I can create one.

> Should we put some info into Changelog when releasing?

Sure

**Review by yilmazbahadir — APPROVED** [review]

**Review by AnthonyLaw — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (0x6861746366574) [reply README.md:6]: Agreed.

Comment (OlegMakarenko) [reply README.md:6]: Added.

File: package.json (hunk 5 above)
Comment (OlegMakarenko) [reply package.json]: Updated.

**Review by 0x6861746366574 — APPROVED** [review]

**Pull request merged**